templates/vdbench_save.yml

```diff
@@ -150,16 +151,38 @@ steps:
       --query 'value[0].message' --output tsv |& tee ${cmdlog}
     grep -q ${cmdokmsg} ${cmdlog} || check_rc.sh 1 "Run azcopy .tar to storage account"
 
+    az resource lock delete -n DO_NOT_DELETE --resource ${STOR_ACCT_RESOURCE_ID}
+    check_rc.sh $rc "Delete Lock on storage account" || exit 0
+
     # Private endpoint removal needed so terraform destory will work.
     az network private-endpoint dns-zone-group delete  \
       --endpoint-name ${RESOURCE_GROUP_NAME}-privendpt \
       --name "privatelink.blob.core.windows.net"       \
       --resource-group ${RESOURCE_GROUP_NAME}-network
     check_rc.sh $? "DNS-Zone delete" || exit 0
 
-    az network private-endpoint delete --name ${RESOURCE_GROUP_NAME}-privendpt \
-      --resource-group ${RESOURCE_GROUP_NAME}-network
-    check_rc.sh $? "Private endpoint delete" || exit 0
+    while [ 1 ]; do
+      cmdlog=/tmp/private_endpt_delete.log
+      # Private endpoint removal needed so terraform destory will work.
+      az network private-endpoint delete --name ${RESOURCE_GROUP_NAME}-privendpt \
+        --resource-group ${RESOURCE_GROUP_NAME}-network |& tee ${cmdlog}
+      grep -q "remove the lock" ${cmdlog}
+      if [ $? -ne 1 ]; then
+        echo "Delete Lock"
+        az resource lock delete -n DO_NOT_DELETE --resource ${STOR_ACCT_RESOURCE_ID}
+        check_rc.sh $rc "Delete Lock on storage account" || exit 0
+        continue
+      else
+        echo "Private Endpoint Deleted"
+        break
+      fi
```

Comment:
What if "az network private-endpoint delete" fails for non-lock reasons? Currently, in that case, this loop will output "Private Endpoint Deleted," but that might not be accurate.

Reply:
updated from convo yesterday